components/role-mgt/org.wso2.carbon.identity.role.v2.mgt.core/src/main/java/org/wso2/carbon/identity/role/v2/mgt/core/RoleManagementServiceImpl.java

```diff
@@ -0,0 +1,772 @@
+
+        try {
+            List<Scope> scopes = RoleManagementServiceComponentHolder.getInstance()
+                    .getApiResourceManager().getScopesByTenantDomain(tenantDomain, "");
+            List<String> scopeNameList = new ArrayList<>();
+            for (Scope scope : scopes) {
+                scopeNameList.add(scope.getName());
+            }
+            for (Permission permission : permissions) {
+
+                if (!scopeNameList.contains(permission.getName())) {
+                    throw new IdentityRoleManagementClientException(INVALID_PERMISSION.getCode(),
+                            "Permission: " + permission.getName() + " not found");
+                }
+            }
+        } catch (APIResourceMgtException e) {
+            throw new IdentityRoleManagementException("Error while retrieving scopes", "Error while retrieving scopes "
+                    + "for tenantDomain: " + tenantDomain, e);
+        }
+    }
+
+    /**
+     * Check if the role name has a domain separator character.
+     * @param roleName Role name.
+     * @return True if the role name has a domain separator character.
+     */
+    private boolean isDomainSeparatorPresent(String roleName) {
+
+        return roleName.contains(UserCoreConstants.DOMAIN_SEPARATOR);
+    }
+
+    /**
+     * Remove similar permissions.
+     *
+     * @param arr1 Array of permissions.
+     * @param arr2 Array of permissions.
+     */
+    private void removeSimilarPermissions(List<Permission> arr1, List<Permission> arr2) {
+        List<Permission> toRemove = new ArrayList<>();
+
+        for (Permission p1 : arr1) {
+            for (Permission p2 : arr2) {
+                if (p1.getName().equals(p2.getName())) {
+                    toRemove.add(p1);
+                    break;
+                }
+            }
+        }
+        arr1.removeAll(toRemove);
+        arr2.removeAll(toRemove);
+    }
+
+    /**
+     * Remove similar idp groups.
+     *
+     * @param arr1 Array of idp groups.
+     * @param arr2 Array of idp groups.
+     */
+    private void removeSimilarIdpGroups(List<IdpGroup> arr1, List<IdpGroup> arr2) {
+        List<IdpGroup> toRemove = new ArrayList<>();
+
+        for (IdpGroup p1 : arr1) {
+            for (IdpGroup p2 : arr2) {
+                if (p1.getGroupId().equals(p2.getGroupId())) {
+                    toRemove.add(p1);
+                    break;
+                }
+            }
+        }
+        arr1.removeAll(toRemove);
+        arr2.removeAll(toRemove);
+    }
+
+    /**
+     * Get the filter node as a list.
+     *
+     * @param filter Filter string.
+     * @throws IdentityRoleManagementException Error when validate filters.
+     */
+    private List<ExpressionNode> getExpressionNodes(String filter) throws IdentityRoleManagementException {
+
+        List<ExpressionNode> expressionNodes = new ArrayList<>();
+        filter = StringUtils.isBlank(filter) ? StringUtils.EMPTY : filter;
```

Comment:
```suggestion
```

Reply:
if the filter value is null, we need set the filter value and empty string